test/extensions/transport_sockets/tls/ssl_socket_test.cc

```diff
@@ -599,6 +599,14 @@ const std::string testUtilV2(const TestUtilOptionsV2& options) {
     EXPECT_EQ(1UL, client_stats_store.counter(options.expectedClientStats()).value());
   }
 
+  if (options.expectSuccess()) {
+    EXPECT_EQ("", client_connection->transportFailureReason());
+    EXPECT_EQ("", server_connection->transportFailureReason());
+  } else {
+    EXPECT_NE("", client_connection->transportFailureReason());
```

Comment:
Should we test a specific failure reason here?

Reply:
Done